### PR TITLE
[Docs] Track UTMs to monitor traffic driving sign-ups for YBM

### PR DESCRIPTION
**Pull request by samiahmedsiddiqui**
Marketing Team is on process to Track UTMs that drive traffic to the docs domain with the goal to track conversions/sign ups for YBM. This PR will create cookies for the UTM parameters and pass to the YBM signup.

**Testing Scenario:**
1. Open https://deploy-preview-19672--infallible-bardeen-164bc9.netlify.app/?utm_source=google&utm_medium=paid_search&utm_campaign=docs
2. Navigate to any page and then click on the YBM Signup link (You can easily find it in header under "Start Now" button)
3. YBM Signup should looks like this: https://cloud.yugabyte.com/signup?utm_source=google&utm_medium=paid_search&utm_campaign=docs

### CODE REVIEW

**Review by premkumr — APPROVED** [review]
Most of these are coding comments rather than logic . it is difficult for me to check/test the logic.

On a side note , we should use js libraries like `lodash`. Will simplify the code.

File: docs/src/track-utm.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      paramsArray.forEach((param) => {
+        const splittedParam = param.split('=');
+        if (splittedParam[0] && splittedParam[1]) {
+          params[splittedParam[0]] = splittedParam[1];
+        }
+      });
+    }
+
+    return params;
+  }
```
[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js:60]: @premkumr No, we cannot use `URLSearchParams` due to the backward compatibility.
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#browser_compatibility

File: docs/src/track-utm.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+  const utmCampaign = getCookie('utm_campaign');
+  const utmMedium = getCookie('utm_medium');
+  const utmSource = getCookie('utm_source');
+  const utmTerm = getCookie('utm_term');
+  const utmContent = getCookie('utm_content');
+
+  // Fill Form Values.
+  if (utmMedium && utmMedium !== '') {
```
[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js:116]: These checks are removed as it not needed on the docs website.

File: docs/src/track-utm.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  }
+
+  if (utmCondition === false) {
+    const keyValuePairs = getQueryParams(window.location.href);
+    let setUTMs = false;
+
+    // Create Cookie for UTM parameters.
+    Object.keys(keyValuePairs).forEach((property) => {
+      if (keyValuePairs[property] !== '') {
+        if (property === 'utm_medium' || property === 'utm_source' || property === 'utm_campaign' || property === 'utm_term' || property === 'utm_content') {
```
[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js:97]: Condition changed.

[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js:97]: Condition changed, even now we don't need switch statement.

File: docs/src/track-utm.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      });
+
+      const totalParams = Object.assign(currentParams, availableUTMs);
+      if (Object.keys(totalParams).length > 0) {
+        if (anchorUrl.includes('?')) {
+          anchorUrl = anchorUrl.split('?')[0];
+        }
+
+        Object.keys(totalParams).forEach((key, index) => {
+          if (index === 0) {
```
[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js:186]: @premkumr I personally don't like ternary operator here.

File: docs/src/track-utm.js [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+  function setCookie(name, value, monthToLive) {
+    let cookie = `${name}=${encodeURIComponent(value)}`;
+    let saveFor = monthToLive;
+
+    if (typeof monthToLive !== 'number') {
+      saveFor = 3;
+    }
+
+    cookie += `; max-age=${(saveFor * 30 * (24 * 60 * 60))}`;
+    if (location.hostname === 'docs.yugabyte.com') {
```
[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js]: Yes, we can. I applied this change in the recent commit.

File: docs/src/track-utm.js [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+    while (fetchCookie < splittedLength) {
+      checkCookie = splittedCookie[fetchCookie];
+      while (checkCookie.charAt(0)) {
+        if (checkCookie.charAt(0) === ' ') {
+          checkCookie = checkCookie.substring(1);
+        } else {
+          break;
+        }
+      }
```
[the remark this answers is not in the document]
Comment (samiahmedsiddiqui) [reply docs/src/track-utm.js]: @premkumr Change applied.

**Comment by samiahmedsiddiqui** [comment]
> Most of these are coding comments rather than logic . it is difficult for me to check/test the logic.

I have made changes, can you please review again?

I have also added the testing scenario in the PR description.

**Comment by timothy-e** [comment]
Sorry - referenced this issue in my commits instead of #19572

**Pull request merged**